FpsCheck.py

```diff
@@ -365,9 +365,11 @@ def find_invalid_well_known(self, check_sets):
             # have stored
             try:
                 json_schema = self.open_and_load_json(url)
-                schema_fields = set(self.acceptable_fields) & set(
-                    json_schema.keys())
                 curr_fps_set = check_sets[primary]
+                schema_fields = set({field for field in json_schema.keys()
+                                     if field in self.acceptable_fields})
+                schema_fields.update({field for field in self.acceptable_fields
+                        if len(curr_fps_set.relevant_fields_dict[field]) > 0})
```

Comment:
Couple of suggestions here:
1. The `{...}` expression is already a [set comprehension](https://python-reference.readthedocs.io/en/latest/docs/comprehensions/set_comprehension.html), so you don't need to wrap it in `set(...)`.
2. Instead of defining `schema_fields` and then immediately mutating it, we should just define it properly the first time, using a more detailed set comprehension.
3. It's idiomatic in Python to check if a list is empty by implicitly converting it to bool, rather than by checking `len(x) > 0`.

```suggestion
                schema_fields = {
                    field
                    for field in self.acceptable_fields
                    if field in json_schema or curr_fps.set.relevant_fields_dict[field]
                }
```